src/engine/server_iv.c

```diff
@@ -1066,26 +1066,38 @@ _iv_op(struct ds_iv_ns *ns, struct ds_iv_key *key, d_sg_list_t *value,
 static void
 iv_op_ult(void *arg)
 {
-	struct iv_op_ult_arg *ult_arg = arg;
+	struct iv_op_ult_arg	*ult_arg = arg;
+	int			rc;
 
-	D_ASSERT(ult_arg->iv_sync.ivs_mode == CRT_IV_SYNC_LAZY);
 	/* Since it will put LAZY sync in a separate and asynchronous ULT, so
 	 * let's use EAGER mode in CRT to make it simipler.
 	 */
 	ult_arg->iv_sync.ivs_mode = CRT_IV_SYNC_EAGER;
```

Comment:
if originally sync->ivs_mode == CRT_IV_SYNC_NONE, that IV OP got retry (for example DER_GERVER), then the retry will call iv_op_by_ult() -> iv_op_ult() here will force it to work in SYNC_EAGER mode, look that is unnecessary for the sync?